jpos/src/main/java/org/jpos/space/JDBMSpace.java

```diff
@@ -322,13 +323,14 @@ public synchronized V in (Object key) {
      */
     public synchronized V in (Object key, long timeout) {
         Object obj;
-        Instant now = Instant.now();
-        long duration;
+        Duration to = Duration.ofMillis(timeout);
+        Duration now = Duration.ofNanos(System.nanoTime());
+        Duration duration;
         while ((obj = inp (key)) == null &&
-                (duration = Duration.between(now, Instant.now()).toMillis()) < timeout)
+                to.compareTo(duration = Duration.ofNanos(System.nanoTime()).minus(now)) > 0)
```

Comment:
You can't pass `System.nanoTime()` to `Instant` directly since `System.nanoTime()` is not a proper epoch but rather a runtime `Duration`. Basically an `Instant` needs to be passed `NanoClock` for monotonic behavior, but here since we only need to compute relative time we an just use a `Duration`.